tools/wptserve/tests/functional/test_request.py

```diff
@@ -47,6 +80,29 @@ def handler(request, response):
         self.assertEqual(200, resp.getcode())
         self.assertEqual(["12345\n", "abcdef\r\n", "zyxwv"], resp.read().split(" "))
 
+    def test_iter_input_longer_than_buffer(self):
+        old_max_buf = InputFile.max_buffer_size
+        InputFile.max_buffer_size = 10
+
+        @wptserve.handlers.handler
+        def handler(request, response):
+            f = request.raw_input
+            return " ".join(line for line in f)
+
+        route = ("POST", "/test/test_iter", handler)
+        self.server.router.register(*route)
+
+        try:
+            resp = self.request(route[1], method="POST", body="12345\nabcdef\r\nzyxwv")
+        except HTTPError:
+            # Cleanup and fail test
+            InputFile.max_buffer_size = old_max_buf
+            self.assertTrue(False, 'This should not be reached. Server failed.')
+
+        self.assertEqual(200, resp.getcode())
+        self.assertEqual(["12345\n", "abcdef\r\n", "zyxwv"], resp.read().split(" "))
+        InputFile.max_buffer_size = old_max_buf
```

Comment:
Maybe rather than having the exception handler in the middle at all, wrap most of the test in a try/finally block for cleanup.

Reply:
Agreed, that would be a cleaner way to do it. Will push that in a moment